coalib/bearlib/languages/documentation/DocBaseClass.py

```diff
@@ -84,3 +93,95 @@ def process_documentation(self, *args, **kwargs):
         """
         raise NotImplementedError('This function has to be implemented for a '
                                   'documentation bear.')
+
+    def instantiate_padding(self, content, doc_comment_iterator):
+        """
+        Instantiates top_padding and bottom_padding of DocumentationComment
+        instances, which contains the current value of blank lines present
+        before and after the documentation. This also changes the `range` of
+        DocumentationComment according to values of padding.
+
+        :param content:
+            Contains the file content.
+        :param doc_comment_iterator:
+            An updated iterator returning each DocumentationComment found
+            in the content.
+        """
+        doc_comments = []
+        for doc_comment in doc_comment_iterator:
+            top_padding = 0
+            bottom_padding = 0
+            start_index = doc_comment.range.start.line - 2
+            end_index = doc_comment.range.end.line
+            while start_index >= 0 and content[start_index] == '\n':
+                top_padding += 1
+                start_index -= 1
+            while end_index < len(content) and content[end_index] == '\n':
+                # This condition will take place if theres an inline docstring
+                # following documentation.
```

Comment:
`content[end_index] `represents the line after end marker.
yup in this case it wont be true as
`end_index > len(content)`
`bottom_padding` will be instantiated to 0 in this case :)